### PR TITLE
[release 4.5] Bug 1855877: cherry-pick hostname fixes

### PR DESCRIPTION
**Pull request by darkmuggle**
templates: ensure RHCOS node hostnames are mostly valid

This aims to fix all the edge-cases around invalid hostnames. This can include:
     - localhost as the hostname
     - a static hostname larger than 63 characters
     - slow reverse DNS hostname discovery
     - truncated hostname in initramfs discovery that is 64 characters long

Fixes BZ 1855878

Includes commits:
- cb12bb855b7493f9b1a23dbe9ed13d5dbaa70d51
- cf2db93410f23d3a5099ed6e619cb286affff538
- 9ddd20dcbe04e90df97b6abcdf511ceef3ec1170

### CODE REVIEW

**Comment by darkmuggle** [comment]
Handled the cherry-pick by hand since there are three commits that are needed for this fix to work.

**Comment by kikisdeliveryservice** [comment]
Pretty sure this is failing unit tests bc in master we had ignition spec3 but 4.5 and less have ign 2. So you probably have  to figure out what changed in these folders you touched between master/4.5 to make sure it can work with the lower version(that change should hold for 4.3/4.4/4.5).

File: templates/common/_base/files/usr-local-sbin-set-valid-hostname.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+mode: 0755
```
Comment (kikisdeliveryservice) [inline templates/common/_base/files/usr-local-sbin-set-valid-hostname.yaml:1]: You need a filesystem here, it's reqd in ign spec2

Comment (darkmuggle) [reply templates/common/_base/files/usr-local-sbin-set-valid-hostname.yaml:2]: fixed :)

**Comment by kikisdeliveryservice** [comment]
```
 level=fatal msg="failed to initialize the cluster: Working towards 0.0.1-2020-07-21-142823: 87% complete, waiting on authentication"
---
time="2020-07-21T15:30:13Z" level=info msg="Reporting job state \"failed\" with reason \"executing_graph:step_failed:utilizing_lease:executing_test:executing_template\"" 
```

Let's try again

/retest

**Pull request merged**